syft/messaging/message.py

```diff
@@ -270,7 +294,20 @@ def unbufferize(worker: AbstractWorker, protobuf_obj: "OperationMessagePB") -> "
             for pb_id in protobuf_obj.operation.return_ids
         ]
 
-        operation_msg = Operation(command, owner, tuple(args), kwargs, tuple(return_ids))
+        return_placeholders = [
+            sy.serde.protobuf.serde._unbufferize(worker, placeholder)
+            for placeholder in protobuf_obj.operation.return_placeholders
+        ]
+
+        if return_placeholders:
+            if len(return_placeholders) == 1:
+                operation_msg = Operation(
+                    command, owner, tuple(args), kwargs, return_placeholders[0]
+                )
+            else:
+                operation_msg = Operation(command, owner, tuple(args), kwargs, return_placeholders)
+        else:
+            operation_msg = Operation(command, owner, tuple(args), kwargs, tuple(return_ids))
```

Comment:
This is a workaround for the overloading of `return_ids` to also contain placeholders some of the time, which doesn't work well in Protobuf, because members of a `oneof` field can't be repeated. In order to remove the need for this, I think some refactoring of the `Operation` class is probably in order, but I'd probably defer it to a later PR.

Reply:
I was initially thinking of adding a separate field to the `Operation` class for placeholders, so that either ids or placeholders could be present (but hopefully not both.) The more I think about it though, the more I wonder if it would make sense to introduce separate classes for `AdHocOperations` and `PlannedOperations` (or something like that.)